samples/cpp/benchmark_app/main.cpp

```diff
@@ -322,8 +322,7 @@ int main(int argc, char* argv[]) {
                                                "<dev1>:<nstreams1>,<dev2>:<nstreams2>" +
                                                " or via configuration file.");
                     }
-                } else if (ov_perf_hint == ov::hint::PerformanceMode::UNDEFINED && !device_config.count(key) &&
-                           (FLAGS_api == "async")) {
+                } else if (!isFlagSetInCommandLine("hint") && !device_config.count(key) && (FLAGS_api == "async")) {
```

Comment:
It is incorrect check. By default hint should be set to tput, but if user want to configure device, he must set hint to be none. And if hint == none but nstreams is not set, it should be determined automatically.

Reply:
@nkogteva  what do you mean that if a user wants to configure a device, hint must be none? Surely I can do `-hint throughput -d MULTI:CPU,GPU` or `-hint latency -d CPU` for example?
My understanding (from the docs) is that throughput is default in OpenVINO runtime, so I would expect the same in benchmark_app: if no hint specified, assume throughput hint.

Reply:
@nkogteva If that is the case, then should I modify the Python benchmark_app instead? 
openvino/tools/benchmark_tool/openvino/tools/benchmark_tool/main.py : 173 is a matching line, and it has the following check:
`elif key not in config[device].keys() and args.api_type == "async" and not is_flag_set_in_command_line('hint'):`

Reply:
> @nkogteva If that is the case, then should I modify the Python benchmark_app instead? openvino/tools/benchmark_tool/openvino/tools/benchmark_tool/main.py : 173 is a matching line, and it has the following check: `elif key not in config[device].keys() and args.api_type == "async" and not is_flag_set_in_command_line('hint'):`

Please look at 80779 ticket for C++ and 80902 for python. PR for C++ was just merged, for python I don't see appropriare PR. I suppose fix for python should be done in scope of 80902

Reply:
> @nkogteva what do you mean with if a user wants to configure a device, hint must be none? Surely I can do `-hint throughput -d MULTI:CPU,GPU` or `-hint latency -d CPU` for example? My understanding (from the docs) is that throughput is default in OpenVINO runtime, so I would expect the same in benchmark_app: if no hint specified, assume throughput hint.

By default benchmark_app use tput hint, but if user want to specify some low granularity settings by himself  he should directly turn off hint. Otherwise it is very confusing and hard to explain how this high level (hints) and low level settings (streams, threads, pinning and so on) are combining with each other. Please see tickets which I mentioned.

Reply:
In our documentation we say that this is supported https://docs.openvino.ai/latest/openvino_docs_OV_UG_Performance_Hints.html#combining-the-hints-and-individual-low-level-settings

"Combining the Hints and Individual Low-Level Settings
While sacrificing the portability at a some extent, it is possible to combine the hints with individual device-specific settings. For example, you can let the device prepare a configuration ov::hint::PerformanceMode::THROUGHPUT while overriding any specific value"

Reply:
@helena-intel Yes, product supports it and if user know how to do it, he can combine them in his oun application. But from tool perspective we don't want to allow user doing it. Otherwise user will set hint latency and nstreams simaltenously and that can lead to unpredictable results for him. It is a desicion on tool level.

Reply:
The documentation is clear that you can do this though, and actually recommends that you can test the effects of this  with benchmark_app. I would expect benchmark_app to do exactly as our product supports. It seems a completely expected use case to want to use throughput hint, but limit for example the number of threads on CPU because your application needs to save some threads for other processing.

From the other side: if setting nstreams and latency simulataneously is always a bad idea, shouldn't it be disallowed in OpenVINO Runtime (either throwing an exception or a warning)? If not, if there are usecases for this, shouldn't benchmark_app support it too?

Reply:
@helena-intel 

1. First of all, benchmark_app should not support everything which product can.

2. This tool is used by a lot of users and not all of them are experinced in our product. Very often it is used as probe what OpenVINO can achive. And when first attemp is disappointed they would never back to try another combination of options. That's why default benchmark_app behaviour differs from default OV behaviour, example (BA uses tpu as default)

3. It is better to avoid situation when user shots in his leg just because he can. I agree, that when product behaviour is uncear, we should fix product, but not tools. But hint + low granular setting is working in some cases, so we can't disable it in product. Moreover hint is just hint and plugin can ignore it, so it is unclear whether plugin should raise exception or not in some particular cases (for example, when streams are used in latency mode on CPU). 

4. We had discussed all these problems before susspension with sample team and with Maxim S who was driving hints introduction. And the only one agreement which we have relates benchmark_app behaviour. We would like to remove low granularity options from benchmark_app at all to keep hints only, but it is impossible because customers get used to use them already.

Reply:
@nkogteva I reverted the C++ changes and implemented the checks in Python as suggested in the ticket 80902, can you verify whether or not everything is correct now?